uaclient/snap.py

```diff
@@ -1,18 +1,27 @@
+import http.client
+import json
 import logging
 import re
+import socket
+from collections import namedtuple
 from typing import List, Optional
 
-from uaclient import apt, event_logger, exceptions, messages, system
+from uaclient import apt, event_logger, exceptions, messages, system, util
 
 SNAP_CMD = "/usr/bin/snap"
 SNAP_INSTALL_RETRIES = [0.5, 1.0, 5.0]
 HTTP_PROXY_OPTION = "proxy.http"
 HTTPS_PROXY_OPTION = "proxy.https"
 SNAP_CHANNEL_SHORTEN_VALUE = "…"
+SNAPD_SOCKET_PATH = "/run/snapd.socket"
+SNAPD_SNAPS_API = "/v2/snaps/{}"
 
 event = event_logger.get_event_logger()
 
 
+SnapInfo = namedtuple("SnapInfo", ("name", "revision", "channel"))
+
+
```

Comment:
We already have `SnapPackage` defined below - we should probably use that, but add types (and convert to a `NamedTuple` if you prefer)